flow-typed/redux.js

```diff
@@ -0,0 +1,74 @@
+declare module 'redux' {
+
+  /*
+
+    S = State
+    A = Action
+
+  */
+
+  declare type Dispatch<S, A> = (action: A) => A;
+
+  declare type MiddlewareAPI<S, A> = {
+    dispatch: Dispatch<S, A>;
+    getState(): S;
+  };
+
+  declare type Store<S, A: { type: $Subtype<string> }> = {
+    // rewrite MiddlewareAPI members in order to get nicer error messages (intersections produce long messages)
+    dispatch: Dispatch<S, A>;
```

Comment:
Ah yes, that `S` was an experiment to retain type info in favour of react-redux (I mean the `dispatch` function injected by `connect` in some cases) but I guess is unnecessary. Better to discard it and simplify both libdefs, thanks for pointing out.